packages/frontend-api/src/Model/Token/TokenFacade.php

```diff
@@ -0,0 +1,131 @@
+use Lcobucci\JWT\Token;
+use Lcobucci\JWT\ValidationData;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser;
+use Shopsys\FrontendApiBundle\Model\Token\Exception\ExpiredTokenUserMessageException;
+use Shopsys\FrontendApiBundle\Model\Token\Exception\InvalidTokenUserMessageException;
+use Shopsys\FrontendApiBundle\Model\Token\Exception\NotVerifiedTokenUserMessageException;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+
+class TokenFacade
+{
+    protected const ACCESS_TOKEN_EXPIRATION = 3600;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Domain\Domain
+     */
+    protected $domain;
+
+    /**
+     * @var \Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface
+     */
+    protected $parameterBag;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface $parameterBag
+     */
+    public function __construct(
+        Domain $domain,
+        ParameterBagInterface $parameterBag
+    ) {
+        $this->domain = $domain;
+        $this->parameterBag = $parameterBag;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser $customerUser
+     * @return string
+     */
+    public function generateAccessTokenByCustomerUser(CustomerUser $customerUser): string
+    {
+        return $this->generateTokenByCustomerUserAndExpiration($customerUser, static::ACCESS_TOKEN_EXPIRATION);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser $customerUser
+     * @param int $expiration
+     * @return string
+     */
+    protected function generateTokenByCustomerUserAndExpiration(CustomerUser $customerUser, int $expiration): string
+    {
+        $currentTime = time();
+
+        $tokenBuilder = (new Builder())
+            ->issuedBy($this->domain->getUrl())
+            ->permittedFor($this->domain->getUrl())
+            ->issuedAt($currentTime)
+            ->expiresAt($currentTime + $expiration);
+
+        foreach (TokenCustomerUserTransformer::transform($customerUser) as $key => $value) {
+            $tokenBuilder->withClaim($key, $value);
+        }
+
+        return (string)$tokenBuilder->getToken($this->getSigner(), $this->getKey());
+    }
+
+    /**
+     * @return \Lcobucci\JWT\Signer\Key
+     */
+    public function getKey(): Key
+    {
+        return new Key($this->parameterBag->get('secret'));
+    }
+
+    /**
+     * @return \Lcobucci\JWT\Signer
+     */
+    public function getSigner(): Signer
+    {
+        return new Signer\Hmac\Sha256();
+    }
+
+    /**
+     * @param string $tokenString
+     * @return \Lcobucci\JWT\Token
+     */
+    public function getTokenByString(string $tokenString): Token
+    {
+        try {
+            return (new Parser())->parse($tokenString);
+        } catch (\Throwable $throwable) {
```

Comment:
_nitpick_ 🤓 
Shouldn't it be just `Throwable` with `use Throwable` at the beginning? (same with BadMethodCallException on line 127)